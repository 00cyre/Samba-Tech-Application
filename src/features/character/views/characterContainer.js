import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import CharacterPage from './characterPage';
import * as characterOperations from '../redux/characterOperations';

export class CharacterContainer extends PureComponent {
  async componentDidMount() {
    await this.props.getCharacter();
  }
  render() {
    return <CharacterPage {...this.props} />
  }
}

const mapDispatchToProps = {
  getCharacter: characterOperations.getCharacters,
};

CharacterContainer.defaultProps = {
    Characters: [],
};

export const mapStateToProps = (state) => {
  return {
    Characters: state.characterReducer
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CharacterContainer);
