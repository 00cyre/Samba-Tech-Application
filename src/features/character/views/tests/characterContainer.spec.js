import React from 'react';
import { CharacterContainer, mapStateToProps } from '../characterContainer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
describe('Component <CharacterContainer/>', function () {
    let component;
    let getCharacter = jest.fn();
    let Characters = [{ name: "Walter White" }];
    beforeEach(() => {
        component = shallow(
            <CharacterContainer
                getCharacter={getCharacter}
                Characters={Characters}
            />
        );
    });

    it('Should Render <CharacterContainer/> ', () => {
        expect(component).toMatchSnapshot();
    });

    it('Should Render  <CharacterPage/> on <CharacterContainer/>', () => {
        expect(component.find('#characterPage').exists()).toEqual(true);
    });
    it('Should Test mapStateToProps', () => {
        expect(mapStateToProps({
            characterReducer: 'lmao'
        })).toEqual({
            Characters: 'lmao'
        });
    });
});
