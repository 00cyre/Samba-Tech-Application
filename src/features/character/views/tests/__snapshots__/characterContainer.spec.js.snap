// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <CharacterContainer/> Should Render <CharacterContainer/>  1`] = `
<WithStyles(CharacterPage)
  Characters={
    Array [
      Object {
        "name": "Walter White",
      },
    ]
  }
  getCharacter={[MockFunction]}
  id="characterPage"
/>
`;
